### ISSUE TITLE
NUTS 2 v NUTS 3 boundaries

### ISSUE THREAD

**Issue by kmmmorris** [rOpenGov/eurostat#80]
Following the very useful [tutorial](http://ropengov.github.io/eurostat/articles/eurostat_tutorial.html)

In the first mapping exercise ("Disposable income of private households by NUTS 2 regions at 1:60mln resolution using ggplot2" the code  plots NUTS 2 level data onto NUTS 3 boundaries. Is a NUTS 2 level shape file available via the Eurostat package and if so how can I load it?

Many thanks for this excellent package.

**Comment by antagomir** [comment]
Thanks! 

Function  [get_eurostat_geospatial.R](https://github.com/rOpenGov/eurostat/blob/master/R/get_eurostat_geospatial.R) can be used to download all resolutions of the [NUTS2013 maps](http://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units), except the 3:1 (which we should add). The function returns SpatialPolygonDataFrames which is the R presentation for shape files. 

As far as I can see, those geographic resolutions themselves are different from NUTS (2/3) resolutions which are based on population densities. Therefore, you can download maps with geographic resolutions 1/10/20/60, and NUTS2/3 demographic data can be mapped on these.with [merge_eurostat_geodata.R](https://github.com/rOpenGov/eurostat/blob/master/R/merge_eurostat_geodata.R).

Does this help ?

**Comment by muuankarski** [comment]
Thanks @kmmmorris for asking and @antagomir for reminding!

Basically there is only one shapefile that has all the different NUTS levels (0-3) in it. You can get the level you want by subsetting the object you get from `get_eurostat_geospatial()`. There are simple reproducible examples below showing the logic using both `data.frames` with `ggplot2`-package and `SpatialPolygonDataFrame` with `sp`-package. 



```r
library(dplyr)
library(ggplot2)
library(eurostat)

# Using data.frame and ggplot2-package
# ----------------------------------
df60 <- get_eurostat_geospatial(output_class = "df", resolution = 60)

# NUTS 0
df60 %>% 
  filter(STAT_LEVL_ == 0) %>% 
  ggplot(., aes(x=long,y=lat,group=group)) +
  geom_polygon(fill=NA,color="black")

# NUTS 1
df60 %>% 
  filter(STAT_LEVL_ == 1) %>% 
  ggplot(., aes(x=long,y=lat,group=group)) +
  geom_polygon(fill=NA,color="black")

# NUTS 2
df60 %>% 
  filter(STAT_LEVL_ == 2) %>% 
  ggplot(., aes(x=long,y=lat,group=group)) +
  geom_polygon(fill=NA,color="black")

# NUTS 3
df60 %>% 
  filter(STAT_LEVL_ == 3) %>% 
  ggplot(., aes(x=long,y=lat,group=group)) +
  geom_polygon(fill=NA,color="black")
```


```r
# Using SpatialPolygonDataFrame-object and sp-package
# ----------------------------------
library(dplyr)
library(eurostat)
library(sp)

spdf60 <- get_eurostat_geospatial(output_class = "spdf", resolution = 60)

# NUTS 0
spdf60[spdf60$STAT_LEVL_ == 0,] %>% plot(.)

# NUTS 1
spdf60[spdf60$STAT_LEVL_ == 1,] %>% plot(.)

# NUTS 2
spdf60[spdf60$STAT_LEVL_ == 2,] %>% plot(.)

# NUTS 3
spdf60[spdf60$STAT_LEVL_ == 3,] %>% plot(.)
``` 

Finally you can see the differences between levels with this simple faceted ggplot2-plot

```r
# As a faceted ggplot-plot
ggplot(df60, aes(x=long,y=lat,group=group)) +
  geom_polygon(fill=NA,color="black") + 
  facet_wrap(~STAT_LEVL_)
```
![rplot](https://cloud.githubusercontent.com/assets/1525240/26104496/c61a227a-3a46-11e7-84a9-d5ebac400c28.png)

Is this of any help to your issue?

**Comment by muuankarski** [comment]
In addition, argument `parameter` in function  `get_eurostat_geospatial()` defines ONLY the resolution of the polygons ie. how detailed they are (and how large is the object). It has nothing to do with NUTS levels.

All shapefiles at all resolutions (1:60million, 1:20million, 1:10million and  1:1million) do contain all the NUTS-levels.

**Comment by antagomir** [comment]
Excellent clarification. I think it would be useful to write this more explicitly in the vignette and function documentation. Hence let us leave this issue open for now.

**Comment by antagomir** [comment]
Related to #81 - also note that we closed #82  and now the get_eurostat_geospatial returns sf object which makes manipulations more straightfwd. The merge_eurostat_geospatial has been hidden. @muuankarski any chance to close this issue and perhaps make the relevant updates in the vignette before I update CRAN? Meanwhile I am working through the other issues.

**Comment by muuankarski** [comment]
Subsetting the geospatial data at right NUTS-level is even more straightforward now with sf-object implemented. Consider there two examples where we are downloading two datasets that are provided at different NUTS-levels, first at NUTS2-level and second at NUTS0-level (nation state). *Both* can be joined with correct geodata with identical `inner_join()` call.

```r
# Disposable income of private households by NUTS 2 regions
# ==============================
euro_sf2 <- get_eurostat("tgs00026", time_format = "raw",
                         stringsAsFactors = FALSE,
                         filter = list(time = "2014"))

# inner join with 
inner_join(get_eurostat_geospatial(resolution = "60", nuts_level = "all"), euro_sf2) %>% 
  select(values) %>% plot()
```

```r
# At-risk-of-poverty thresholds - EU-SILC survey
# ==============================
poverty_thresholds <- get_eurostat("ilc_li01", time_format = "raw",
                                     stringsAsFactors = FALSE,
                                     filter = list(time = "2014", 
                                                   indic_il = "LI_C_M60", 
                                                   currency = "EUR",
                                                   hhtyp = "A1"))


inner_join(get_eurostat_geospatial(resolution = "60", nuts_level = "all"), poverty_thresholds) %>% 
  select(values) %>% plot()
```
